File: LeetCode/35.Search_Insert_Position/main.cpp
#include "solution.h"

// Unit Test

int main(){
    Solution s;
    vector<int> v1 {1,3,5,6};
    int t1 = 5;
    int t2 = 2;
    int t3 = 7;
    vector<int> vextra {1};
    int textra = 0;
    cout << "First Function Test Start" << endl;
    if (s.searchInsert(v1, t1) == 2) { cout << "Pass" << endl;}
    else { cout << "Failed 1 - v1 t1" << endl;}
    if (s.searchInsert(v1, t2) == 1) { cout << "Pass" << endl;}
    else { cout << "Failed 2 - v1 t2" << endl;}
    if (s.searchInsert(v1, t3) == 4) { cout << "Pass" << endl;}
    else { cout << "Failed 3 - v1 t3" << endl;}
    if (s.searchInsert(v1, textra) == 0) { cout << "Pass" << endl;}
    else { cout << "Failed 4 - v1 textra" << endl;}
    if (s.searchInsert(vextra, textra) == 0) { cout << "Pass" << endl;}
    else { cout << "Failed extra - vextra textra" << endl;}
    cout << "First Function Test End" << endl;
    cout << "------------------------" << endl;
    cout << "Second Function Test Start" << endl;
    if (s.searchInsert2(v1, t1) == 2) { cout << "Pass" << endl;}
    else { cout << "Second Failed 1 - v1 t1" << endl;}
    if (s.searchInsert2(v1, t2) == 1) { cout << "Pass" << endl;}
    else { cout << "Second Failed 2 - v1 t2" << endl;}
    if (s.searchInsert2(v1, t3) == 4) { cout << "Pass" << endl;}
    else { cout << "Second Failed 3 - v1 t3" << endl;}
    if (s.searchInsert2(v1, textra) == 0) { cout << "Pass" << endl;}
    else { cout << "Second Failed 4 - v1 textra" << endl;}
    if (s.searchInsert2(vextra, textra) == 0) { cout << "Pass" << endl;}
    else { cout << "Second Failed extra - vextra textra" << endl;}
    cout << "Second Function Test End" << endl;
    return 0;
}

File: LeetCode/35.Search_Insert_Position/main.py
#!/usr/bin/python

from solution import Solution

# Unit Test
nums1 = [1,3,5,6]
nums2 = [1]
t1 = 5
t2 = 2
t3 = 7 
t4 = 0

s1 = Solution(nums1, t1)
s2 = Solution(nums1, t2)
s3 = Solution(nums1, t3)
s4 = Solution(nums1, t4)
s5 = Solution(nums2, t4)

if (s1.searchInsert()==2):
    print("Pass")
    s1.resetAll()
else:
    print("Test 1 Failed")
if (s2.searchInsert()==1):
    print("Pass")
    s2.resetAll()
else:
    print("Test 2 Failed")
if (s3.searchInsert()==4):
    print("Pass")
    s3.resetAll()
else:
    print("Test 3 Failed")
if (s4.searchInsert()==0):
    print("Pass")
    s4.resetAll()
else:
    print("Test 4 Failed")
if (s5.searchInsert()==0):
    print("Pass")
    s5.resetAll()
else:
    print("Test 5 Failed")